### ISSUE TITLE
Data not accessible through head methods in defineNuxtComponent

### ISSUE THREAD

**Issue by Alex94800** [nuxt/framework#6006]
### Environment

- Operating System: `Darwin`
- Node Version:     `v15.5.0`
- Nuxt Version:     `3.0.0-rc.6`
- Package Manager:  `yarn@1.22.10`
- Builder:          `vite`
- User Config:      `head`, `modules`, `buildModules`, `css`, `build`
- Runtime Modules:  `@vueuse/motion/nuxt@2.0.0-beta.18`, `@nuxtjs/strapi@1.5.0`
- Build Modules:    `-`

### Reproduction

https://stackblitz.com/github/nuxt/starter/tree/v3-stackblitz?file=app.vue

### Describe the bug

The title is not filled with the variable, but it works when applying a title without variable.

**Comment by danielroe** [comment]
This is not a bug. `head` receives `nuxtApp` as an argument but does not have access to the _this_ context of the component.

**Comment by Alex94800** [comment]
@danielroe Ok I get it, but _this_ was accessible in Nuxt 2. How can we access data in the head method then ? I can't find anything in the documentation.

**Comment by mmis1000** [comment]
> @danielroe Ok I get it, but _this_ was accessible in Nuxt 2. How can we access data in the head method then ? I can't find anything in the documentation.

https://v3.nuxtjs.org/guide/features/head-management#meta-components

I guess you just use this instead?

**Comment by ennioVisco** [comment]
> @danielroe Ok I get it, but _this_ was accessible in Nuxt 2. How can we access data in the head method then ? I can't find anything in the documentation.

I'm sorry, I'm not familiar with nuxt 2, but I'm pretty sure that the `useHead` composable pretty much does the same thing. Don't you agree? (see https://v3.nuxtjs.org/guide/features/head-management#usehead-composable)

**Comment by Alex94800** [comment]
@ennioVisco you are absolutely right, but it would require to use the composition API that I don't like at all

**Comment by danielroe** [comment]
You can use the meta components: https://v3.nuxtjs.org/guide/features/head-management/#meta-components.

**Comment by Alex94800** [comment]
@danielroe yeah I used this method instead, thank you
